lib/services/batch/lib/batchSharedKeyCredentials.js

```diff
@@ -0,0 +1,168 @@
+
+/**
+* Creates a new BatchSharedKeyCredentials object.
+* @constructor
+* @param {string} accountName The batch account name. 
+* @param {string} accountKey The batch account key.
+*/
+function BatchSharedKeyCredentials(accountName, accountKey) {
+  if (!Boolean(accountName) || typeof accountName.valueOf() !== 'string') {
+    throw new Error('accountName must be a non empty string.');
+  }
+    
+  if (!Boolean(accountKey) || typeof accountKey.valueOf() !== 'string') {
+    throw new Error('accountKey must be a non empty string.');
+  }
+    
+  this.accountName = accountName;
+  this.accountKey = accountKey;
+  this.signer = new HmacSha256Sign(accountKey);
+}
+
+/**
+* Signs a request with the Authentication header.
+*
+* @param {webResource} The WebResource to be signed.
+* @param {function(error)}  callback  The callback function.
+* @return {undefined}
+*/
+BatchSharedKeyCredentials.prototype.signRequest = function (webResource, callback) {
+  var self = this;
+    
+  var getvalueToAppend = function (value, headerName) {
+    if (!Boolean(value) || !Boolean(value[headerName])) {
+      return '\n';
+    } else {
+      return value[headerName] + '\n';
+    }
+  };
+  
+  var getContentLengthToAppend = function (value, method, body) {
+    if (!Boolean(value) || !Boolean(value['Content-Length'])) {
+      if (body) {
+        return Buffer.byteLength(body) + '\n';
+      }
+      if (method === 'GET') {
+        return '\n';
+      } else {
+        return '0\n';
+      }
+    } else {
+      return value['Content-Length'] + '\n';
+    }
+  };
+    
+  // Set Headers
+  if (!Boolean(webResource.headers['ocp-date'])) {
+      webResource.headers['ocp-date'] = new Date().toUTCString();
+  }
+
+  var stringToSign =
+      webResource.method + '\n' +
+  getvalueToAppend(webResource.headers, 'Content-Encoding') +
+  getvalueToAppend(webResource.headers, 'Content-Language') +
+  getContentLengthToAppend(webResource.headers, webResource.method, webResource.body) +
+  getvalueToAppend(webResource.headers, 'Content-MD5') +
+  getvalueToAppend(webResource.headers, 'Content-Type') +
+  getvalueToAppend(webResource.headers, 'Date') +
+  getvalueToAppend(webResource.headers, 'If-Modified-Since') +
+  getvalueToAppend(webResource.headers, 'If-Match') +
+  getvalueToAppend(webResource.headers, 'If-None-Match') +
+  getvalueToAppend(webResource.headers, 'If-Unmodified-Since') +
+  getvalueToAppend(webResource.headers, 'Range') +
+  this._getCanonicalizedHeaders(webResource) +
+  this._getCanonicalizedResource(webResource);
+    
+  var signature = this.signer.sign(stringToSign);
+    
+  webResource.headers[Constants.HeaderConstants.AUTHORIZATION] = 
+    util.format('SharedKey %s:%s', self.accountName, signature);
+  callback(null);
+};
+
+/*
+* Constructs the Canonicalized Headers string.
+*
+* To construct the CanonicalizedHeaders portion of the signature string,
+* follow these steps: 1. Retrieve all headers for the resource that begin
```

Comment:
This looks like a manually written file.
Please make sure that every step is documented on a new line